lib/gtl_formatter.rb

```diff
@@ -45,6 +57,7 @@ class GtlFormatter
   ].freeze
 
   def self.format_cols(view, row, controller, options)
+
```

Comment:
As an aside,  for "future", I cannot see why the `state`, `hardware`, and `special_cases` below  are defined in the method instead as constants.  It looks like we only read from `special_cases` on line 99-102.  We're redefining the same objects everytime this method is called.

Also, we're sharing the same `state` and `hardware`  hash in several keys so if those hashes are modified, several key's values will change.